src/me/mrCookieSlime/Slimefun/SlimefunGuide.java

```diff
@@ -726,21 +713,107 @@ public void run(final Player p) {
 			}
 			
 			menu.open(p);
-		}		
+		}
 
 		if (survival) {
 			addToHistory(p, category);
 		}
 	}
 
-	public static void addToHistory(Player p, Object obj) {
-		LinkedList<Object> list = getHistory().get(p.getUniqueId());
-		
-		if (list == null) {
-			list = new LinkedList<>();
-			getHistory().put(p.getUniqueId(), list);
+	public static void openSearch(Player player, String searchTerm, boolean cheat, boolean addToHistory) {
+		final ChestMenu menu = new ChestMenu("Slimefun Guide Search");
+
+		menu.setEmptySlotsClickable(false);
+
+		fillInv(menu, cheat);
+
+		addBackButton(menu, player, false, cheat);
+
+		searchTerm = searchTerm.toLowerCase();
+
+		int index = 9;
+		// Find items and add them
+		for (SlimefunItem item : SlimefunItem.list()) {
+			final String itemName = ChatColor.stripColor(item.getItem().getItemMeta().getDisplayName()).toLowerCase();
+
+			if (itemName.isEmpty()) continue;
+
+			if (index == 44) break;
+
+			if (itemName.equals(searchTerm) || itemName.contains(searchTerm)) {
+				menu.addItem(index, item.getItem());
+				menu.addMenuClickHandler(index, (pl, slot, itm, action) -> {
+					if (cheat)
+						pl.getInventory().addItem(itm);
+					else
+						displayItem(pl, itm, true, false, 0);
+					return false;
+				});
+
+				index++;
+			}
 		}
-		
+
+		if (addToHistory)
+			addToHistory(player, searchTerm);
+
+		menu.open(player);
+	}
+
+	private static void fillInv(ChestMenu menu, boolean cheat) {
+		for (int i = 0; i < 9; i++) {
+			menu.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "));
+			menu.addMenuClickHandler(i, (arg0, arg1, arg2, arg3) -> false);
+		}
+
+		// Search feature!
+		menu.addItem(7, new CustomItem(Material.NAME_TAG, "&7Search",
+			"&7Search directly for items"));
+		menu.addMenuClickHandler(7, (player, i, itemStack, clickAction) -> {
+			player.closeInventory();
+			player.sendMessage(ChatColor.AQUA + "What would you like to search for?");
```

Comment:
I think this String should be added to Slimefun's localization file.
And maybe add another line saying "Enter your search term in chat" or whatever?
To make it clear for everyone.